src/config.rs

```diff
@@ -1459,6 +1459,101 @@ pub struct Delta<T: Default> {
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("struct Padding")
+            }
+
+            fn visit_map<V>(self, mut map: V) -> ::std::result::Result<Padding, V::Error>
+                where V: MapAccess<'de>
+            {
+                let mut top = None;
+                let mut right = None;
+                let mut bottom = None;
+                let mut left = None;
+                let mut x = None;
+                let mut y = None;
+                while let Some(key) = map.next_key()? {
+                    match key {
+                        Field::Top => {
+                            if top.is_some() {
+                                return Err(de::Error::duplicate_field("top"));
+                            }
+                            top = Some(map.next_value()?);
+                        }
+                        Field::Right => {
+                            if right.is_some() {
+                                return Err(de::Error::duplicate_field("right"));
+                            }
+                            right = Some(map.next_value()?);
+                        }
+                        Field::Bottom => {
+                            if bottom.is_some() {
+                                return Err(de::Error::duplicate_field("bottom"));
+                            }
+                            bottom = Some(map.next_value()?);
+                        }
+                        Field::Left => {
+                            if left.is_some() {
+                                return Err(de::Error::duplicate_field("left"));
+                            }
+                            left = Some(map.next_value()?);
+                        }
+                        Field::X => {
+                            if x.is_some() {
+                                return Err(de::Error::duplicate_field("x"));
+                            }
+                            x = Some(map.next_value()?);
+                        }
+                        Field::Y => {
+                            if y.is_some() {
+                                return Err(de::Error::duplicate_field("y"));
+                            }
+                            y = Some(map.next_value()?);
+                        }
+                    }
+                }
+                let default_value = 2;
+                let top = top.unwrap_or(y.unwrap_or(default_value));
+                let right = right.unwrap_or(x.unwrap_or(default_value));
+                let bottom = bottom.unwrap_or(y.unwrap_or(default_value));
+                let left = left.unwrap_or(x.unwrap_or(default_value));
+                Ok(Padding{ top, right, bottom, left})
+            }
+        }
+
+        const FIELDS: &'static [&'static str] = &["top", "right", "bottom", "left", "x", "y"];
+        deserializer.deserialize_struct("Duration", FIELDS, PaddingVisitor)
+
+    }
+}
+
```

Comment:
This seems far more complicated than necessary. Why didn't you just add the `x` and `y` field to your current `Padding` struct? Then you could just check if those are set and emit a deprecation warning and update top/right/bottom/left.

Reply:
Oh, I was thinking we could just leave the `x` and `y` fields since most people only care about those anyway. And generally, even if people care about `top`, they could just set:
```
  x: 2
  top: 25
  bottom: 2
```

or whatever. But if we want to just fully migrate over to the new format, I can do that.

Reply:
I think the redundanc of having both `y` and `top/bottom` would confuse people. Keeping around `y` as an option seems like a viable path, but I feel like it just creates unnecessary complexity where the user might have `y` + `top/bottom` set and we run into a scenario where the result is not directly obvious (unless explicitly documented).

So for the sake of simplifying the code and having just one way to do things instead of two, I'd rather see a complete migration with deprecation of `x/y` instead of a mix of the two. That's just my personal opinion though.

Reply:
ok, I'll try to get that done tonight